apiserver/facades/agent/machine/machiner.go

```diff
@@ -155,31 +164,67 @@ func (api *MachinerAPI) SetMachineAddresses(ctx context.Context, args params.Set
 	return results, nil
 }
 
-// Jobs returns the jobs assigned to the given entities.
-func (api *MachinerAPI) Jobs(ctx context.Context, args params.Entities) (params.JobsResults, error) {
-	result := params.JobsResults{
+// SetMachineAddresses is not supported in MachinerAPI at version 5.
+func (api *MachinerAPIv5) SetMachineAddresses(ctx context.Context, args params.SetMachinesAddresses) (params.ErrorResults, error) {
+	return params.ErrorResults{}, errors.NotSupported
+}
+
+// Jobs is not supported in MachinerAPI at version 5.
+// Deprecated: Jobs is being deprecated. Use IsController instead.
+func (api *MachinerAPIv5) Jobs(ctx context.Context, args params.Entities) (params.JobsResults, error) {
+	results := params.JobsResults{
 		Results: make([]params.JobsResult, len(args.Entities)),
 	}
 
-	canRead, err := api.getCanRead()
-	if err != nil {
-		return result, err
+	for i, entity := range args.Entities {
+		isController, err := api.machineService.IsMachineController(ctx, machine.Name(entity.Tag))
+		if err != nil {
+			results.Results[i].Error = apiservererrors.ServerError(err)
+			continue
+		}
+		var jobs []string
+		if isController {
+			jobs = []string{"api-server", "host-units"}
+		} else {
+			jobs = []string{"host-units"}
```

Comment:
You could instantiate with this and just append if it's a controller. No biggie, but saves a block.